accepted/2021/winforms/streamline-application-bootstrap.md

```diff
@@ -0,0 +1,516 @@
+way, you avoid having to explain the same thing over and over again.
+-->
+
+### Why store settings as MSBuild properties (e.g. in csproj or props files)?
+
+
+After deliberations and discussions we propose the settings to be stored as MSBuild properties for the following reasons:
+
+- The designer surface process fun in Visual Studio should have no issues reading MSBuild properties.
+- Unlike schema-enforced app.config, MSBuild properties are stored in an untyped propertybag.
+- app.configs don't appear to be used in .NET apps.
+- Whilst developers who build their apps on .NET Framework 4.7+ maybe familiar with [app.config ApplicationConfigurationSection](https://docs.microsoft.com/dotnet/framework/configure-apps/file-schema/winforms/windows-forms-add-configuration-element) and/or [app.config AppContextSwitchOverrides](https://docs.microsoft.com/dotnet/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element), we have removed all dependencies on app.config and these quirks in .NET Core 3.0 timeframe.
+  > :thought_balloon:  It is also worth noting that quirks were necessary in .NET Framework, which was installed in-place, and didn't provided any kind of side-by-side deployment. In a rare and unfortunate event where we may accidentally break end users experiences in a given release users will be able to install a previous working .NET release and run their application against it until a fix becomes available.
+- There may also be an argument that app.config can be changed by end users, thus allowing to alter behaviours of applications without a recompilation, making the app.config the prefered configuration vehicle. But it is important to note that our SDK is built with developers in mind, and not the ability of end users to alter the application behaviours at will. If an app developer feels it is important to allow end users to change the default font or dpi settings via an app.config it is the developers responsibility to facilitate that mechanism to end users.
+
+Other storage mechanisms were also considered:
+
+
+### Font Configuration
+
+#### Default scenario
+
+Initial thinking was to allow configuration of only [`Font.FamilyName`](https://docs.microsoft.com/dotnet/api/system.drawing.font.fontfamily) and [`Font.Size`](https://docs.microsoft.com/dotnet/api/system.drawing.font.size) properties. However these two properties may be insufficient in some use cases, which otherwise would be achievable if `Application.SetDefaultFont(Font)` API was invoked manually.
+E.g.:
+
+```cs
+Application.SetDefaultFont(new Font(new FontFamily("Calibri"), 11f, FontStyle.Italic | FontStyle.Bold, GraphicsUnit.Point));
+```
+
+It would rather be impractical to provide an MSBuild properties for each argument the `Font` constructor takes, and instead the proposal is to configure fonts via a single property `ApplicationFont`. This property will have a format equivalent to the output of `FontConverter.ConvertToInvariantString()` API e.g.:
```

Comment:
```suggestion
It would rather be impractical to provide an MSBuild properties for each argument the `Font` constructor takes, and instead the proposal is to configure fonts via a single property `ApplicationFont`. This property will have a format equivalent to the output of [`FontConverter.ConvertToInvariantString()` API](https://github.com/dotnet/runtime/blob/00ee1c18715723e62484c9bc8a14f517455fc3b3/src/libraries/System.Drawing.Common/src/System/Drawing/FontConverter.cs#L29-L86) e.g.:
```